src/components/refactor/dialogs/shunt-compensator-creation/shunt-compensator-creation-form.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import Grid from '@mui/material/Grid';
+import { Box } from '@mui/material';
+import BooleanInput from 'components/refactor/rhf-inputs/boolean-input';
+import IntegerInput from 'components/refactor/rhf-inputs/integer-input';
+import {
+    CURRENT_NUMBER_OF_SECTIONS,
+    EQUIPMENT_ID,
+    EQUIPMENT_NAME,
+    IDENTICAL_SECTIONS,
+    MAXIMUM_NUMBER_OF_SECTIONS,
+    SUSCEPTANCE_PER_SECTION,
+} from 'components/refactor/utils/field-constants';
+import React from 'react';
+import {
+    filledTextField,
+    gridItem,
+    GridSection,
+    SusceptanceAdornment,
+    toPositiveIntValue,
+} from '../../../dialogs/dialogUtils';
+
+import TextInput from '../../rhf-inputs/text-input';
+import { ConnectivityForm } from '../connectivity/connectivity-form';
+import FloatInput from 'components/refactor/rhf-inputs/float-input';
+
+const ShuntCompensatorCreationForm = ({ voltageLevelOptionsPromise }) => {
+    const disabledChecked = { disabled: true };
+
+    const shuntCompensatorIdField = (
+        <TextInput
+            name={EQUIPMENT_ID}
+            label={'ID'}
+            formProps={{ autoFocus: true, ...filledTextField }}
+        />
+    );
+
+    const shuntCompensatorNameField = (
+        <TextInput
+            name={EQUIPMENT_NAME}
+            label={'Name'}
+            formProps={filledTextField}
+        />
+    );
+
+    const maximumNumberOfSectionsField = (
+        <IntegerInput
+            name={MAXIMUM_NUMBER_OF_SECTIONS}
+            label={'ShuntMaximumNumberOfSections'}
+            transformValue={toPositiveIntValue}
```

Comment:
this does not work, transformValue propTypes should be removed from IntegerInput since it does not actually exist
use something like "acceptValue" instead (check FloatInput for an exemple)

Reply:
I added a prop "isPositive",  based on it the desired transformation will be applied